apps/omg_child_chain/lib/omg_child_chain/fees/fee_parser.ex

```diff
@@ -11,35 +11,44 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-
 defmodule OMG.ChildChain.Fees.FeeParser do
   @moduledoc """
   Transaction's fee validation functions
   """
-
   require Logger
 
   @doc """
   Parses and validates json encoded fee specifications file
-
   Parses provided json string to token-fee map and returns the map together with possible parsing errors
   """
-  @spec parse(binary()) :: {:ok, OMG.Fees.fee_t()} | {:error, list({:error, atom()})}
+  @spec parse(binary()) :: {:ok, OMG.Fees.full_fee_t()} | {:error, list({:error, atom()})}
   def parse(file_content) do
     with {:ok, json} <- Jason.decode(file_content) do
-      {errors, token_fee_map, _} =
+      {errors, fee_specs} =
         json
-        |> Enum.map(&parse_fee_spec/1)
-        |> Enum.reduce({[], %{}, 1}, &spec_reducer/2)
+        |> Enum.reduce({[], %{}}, fn {tx_type, fee_spec}, {all_errors, fee_specs} ->
```

Comment:
same as above, this needs simplification with reduced amount of anonymous functions in pipelines